src/seedu/addressbook/commands/Command.java

```diff
@@ -27,6 +27,14 @@ protected Command() {
     }
 
     /**
+     *
+     * @return whether there is a need to save data to file
+     */
+    public boolean isMutating(){
+        return false;
```

Comment:
Having a default implementation for the `isMutating` method helps to reduce the amount of code. However, some may argue that choosing `false` (or `true`) as the default is arbitrary and hurts the intuitiveness of the code. Making the method `abstract` forces all child classes to declare whether it is a mutating command or not, rather than depend on an arbitrary decision made in the parent class.

Reply:
Alright sure. Thanks!